packages/aws-rfdk/lib/lambdas/nodejs/x509-certificate/test/acm-handlers.test.ts

```diff
@@ -0,0 +1,459 @@
+      } catch(e) {
+        error = e;
+      }
+
+      // THEN
+      expect(error).toBeDefined();
+      expect(error.message).toMatch(/Secret .* did not contain a SecretString as expected/);
+      expect(getSecretValueFake.calledOnce).toBe(true);
+    });
+
+    test('retries importing certificate', async () => {
+      // GIVEN
+      const resourceTable = new MockCompositeStringIndexTable();
+      const getItemStub = sinon.stub(resourceTable, 'getItem').resolves(undefined);
+      const putItemStub = sinon.stub(resourceTable, 'putItem').resolves(true);
+
+      const importCertificateStub = sinon.stub()
+        .onFirstCall().rejects('Rate exceeded')
+        .onSecondCall().rejects('Rate exceeded')
+        .onThirdCall().resolves({ CertificateArn: certArn });
+      AWSMock.mock('ACM', 'importCertificate', importCertificateStub);
+
+      const backoffJitterStub = sinon.stub(BackoffGenerator.prototype, 'backoffJitter').resolves();
+      const shouldContinueStub = sinon.stub(BackoffGenerator.prototype, 'shouldContinue').returns(true);
+
+      const importer = new TestAcmCertificateImporter({
+        acm: new AWS.ACM(),
+        dynamoDb: new AWS.DynamoDB(),
+        secretsManager: new AWS.SecretsManager(),
+        resourceTableOverride: resourceTable,
+      });
+
+      // WHEN
+      await expect(importer.doCreate(physicalId, doCreateProps))
+
+      // THEN
+        .resolves.toEqual({ CertificateArn: certArn });
+      expect(getItemStub.calledOnce).toBe(true);
+      expect(putItemStub.calledOnce).toBe(true);
+      expect(importCertificateStub.calledThrice).toBe(true);
+      expect(backoffJitterStub.callCount).toEqual(2);
+      // An additional check is made before logging "Retrying..."
+      expect(shouldContinueStub.callCount).toEqual(4);
+    });
+
+    describe('existing', () => {
+      test('throws if item ARN is missing', async () => {
+        // GIVEN
+        const resourceTable = new MockCompositeStringIndexTable();
+        const getItemStub = sinon.stub(resourceTable, 'getItem').resolves({});
+
+        const importer = new TestAcmCertificateImporter({
+          acm: new AWS.ACM(),
+          dynamoDb: new AWS.DynamoDB(),
+          secretsManager: new AWS.SecretsManager(),
+          resourceTableOverride: resourceTable,
+        });
+
+        // WHEN
+        await expect(importer.doCreate(physicalId, doCreateProps))
+
+        // THEN
+          .rejects.toEqual(new Error("Database Item missing 'ARN' attribute"));
+        expect(getItemStub.calledOnce).toBe(true);
+      });
+
+      test('throws if certificate not found in ACM', async () => {
+        // GIVEN
+        const resourceTable = new MockCompositeStringIndexTable();
+        const getItemStub = sinon.stub(resourceTable, 'getItem').resolves({ ARN: certArn });
+
+        const getCertificateFake = sinon.fake.resolves({});
+        AWSMock.mock('ACM', 'getCertificate', getCertificateFake);
```

Comment:
Does `ACM.getCertificate(...)` return a resolved promise with an empty object when no certificate is found? What is the behavior our logic is using to determine if a certificate exists using this API?

Reply:
Great - thanks for confirming that!